manage-data/data-store/data-streams/failure-store-recipes.md

```diff
@@ -0,0 +1,1153 @@
+        "field": "counter"
+      }
+    },
+    {
+      "set": {
+        "tag": "copy to new counter again",
+        "field": "new_counter",
+        "copy_from": "counter"
+      }
+    }
+  ]
+}
+```
+
+We ingest some data and find that it was sent to the failure store
+```console
+POST my-datastream-ingest/_doc?pipeline=complicated-processor
+{
+    "@timestamp": "2025-04-21T00:00:00Z",
+    "counter_name": "test"
+}
+```
+
+```console-result
+{
+  "_index": ".fs-my-datastream-ingest-2025.05.09-000001",
+  "_id": "HnTJs5YBwrYNjPmaFcri",
+  "_version": 1,
+  "result": "created",
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "failed": 0
+  },
+  "_seq_no": 1,
+  "_primary_term": 1,
+  "failure_store": "used"
+}
+```
+
+Upon checking the failure, we can quickly identify the tagged processor that caused the problem
+```console
+GET my-datastream-ingest::failures/_search
+```
+
+```console-result
+{
+  "took": 0,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": ".fs-my-datastream-ingest-2025.05.09-000001",
+        "_id": "HnTJs5YBwrYNjPmaFcri",
+        "_score": 1,
+        "_source": {
+          "@timestamp": "2025-05-09T06:41:24.775Z",
+          "document": {
+            "index": "my-datastream-ingest",
+            "source": {
+              "@timestamp": "2025-04-21T00:00:00Z",
+              "counter_name": "test"
+            }
+          },
+          "error": {
+            "type": "illegal_argument_exception",
+            "message": "field [counter] not present as part of path [counter]",
+            "stack_trace": """j.l.IllegalArgumentException: field [counter] not present as part of path [counter]
+	at o.e.i.IngestDocument.getFieldValue(IngestDocument.java:202)
+	at o.e.i.c.SetProcessor.execute(SetProcessor.java:86)
+	... 14 more
+""",
+            "pipeline_trace": [
+              "complicated-processor"
+            ],
+            "pipeline": "complicated-processor",
+            "processor_type": "set", // Helpful, but which set processor on the pipeline could it be?
+            "processor_tag": "copy to new counter again" // The tag of the exact processor that it failed on.
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+Without tags in place it would not be as clear where in the pipeline we encountered the problem. Tags provide a unique identifier for a processor that can be quickly referenced in case of an ingest failure.
+
+## Alerting on failed ingestion [failure-store-recipes-alerting]
+
+Since failure stores can be searched just like a normal data stream, we can use them as inputs to [alerting rules](./failure-store.md) in Kibana. Here is a simple alerting example to trigger on more than ten failures in the last five minutes for a data stream:
```

Comment:
```suggestion
Since failure stores can be searched just like a normal data stream, we can use them as inputs to [alerting rules](./failure-store.md) in Kibana. Here is a simple alerting example that is triggered when more than ten indexing failures have occurred in the last five minutes for a data stream:
```
maybe this reads a wee bit better?